metadata-ingestion/src/datahub/ingestion/source/identity/azure_ad.py

```diff
@@ -231,99 +235,101 @@ def _get_azure_group_users(self, azure_group):
             else:
                 error_str = f"Response status code: {str(response.status_code)}. Response content: {str(response.content)}"
                 logger.error(error_str)
-                self.report.report_failure("_get_azure_groups", error_str)
+                self.report.report_failure("_get_azure_ad_groups", error_str)
                 continue
 
-    def _map_azure_groups(self, azure_groups):
-        for azure_group in azure_groups:
-            corp_group_urn = self._map_azure_group_to_urn(azure_group)
+    def _map_azure_ad_groups(self, azure_ad_groups):
+        for azure_ad_group in azure_ad_groups:
+            corp_group_urn = self._map_azure_ad_group_to_urn(azure_ad_group)
             if not corp_group_urn:
                 error_str = "Failed to extract DataHub Group Name from Azure Group for group named {}. Skipping...".format(
-                    azure_group.get("displayName")
+                    azure_ad_group.get("displayName")
                 )
                 logger.error(error_str)
-                self.report.report_failure("azure_group_mapping", error_str)
+                self.report.report_failure("azure_ad_group_mapping", error_str)
                 continue
             corp_group_snapshot = CorpGroupSnapshot(
                 urn=corp_group_urn,
                 aspects=[],
             )
-            corp_group_info = self._map_azure_group_to_corp_group(azure_group)
+            corp_group_info = self._map_azure_ad_group_to_corp_group(azure_ad_group)
             corp_group_snapshot.aspects.append(corp_group_info)
             yield corp_group_snapshot
 
     # Converts Azure group profile into DataHub CorpGroupInfoClass Aspect
-    def _map_azure_group_to_corp_group(self, group):
+    def _map_azure_ad_group_to_corp_group(self, group):
         return CorpGroupInfoClass(
-            displayName=self._map_azure_group_to_group_name(group),
+            displayName=self._map_azure_ad_group_to_group_name(group),
             description=group.get("description"),
-            email=group.get("mail"),
+            email=str(group.get("mail", "")),
```

Comment:
Why was this change needed? 
email seems to be an optional field in the CorpGroupInfoClass. 
```
/**
   * email of this group
   */
  email: optional EmailAddress
```